### ISSUE TITLE
Crash when using remote control API

### ISSUE THREAD

**Issue by Lanthade** [plexinc/plex-media-player#670]
I have a home automation system that issues commands to PMP via the remote control API.  With some regularity PMP crashes.  This seems to have gotten worse lately though I've also been watching more DVR'ed TV and hitting FF on commercials.  The crashing happens while navigating the UI or controlling while playing media.  It is entirely possible that the automation system is sending a badly formed command that is exposing a flaw in the API implementation causing the crash.

PMP 2.5.0.792-1bc7f225 on OSX 10.11.

**Comment by Lanthade** [comment]
I'm crashing 100% of the time when sending the a moveDown command.  Here's the relevant log lines:

2018-03-02 23:08:51 [ DEBUG ] HTTPServer.cpp @ 165 - Incoming request to: "/player/timeline/poll?commandID=0&X-Plex-Device-Name=Q-SYS_Core110f" from "10.0.1.50" 
2018-03-02 23:08:51 [ DEBUG ] HTTPServer.cpp @ 165 - Incoming request to: "/player/navigation/moveDown?commandID=1&X-Plex-Device-Name=Q-SYS_Core110f" from "" 
2018-03-02 23:08:51 [ INFO  ] JS: [Companion] Successfully processed command from path "/player/navigation/moveDown"

**Comment by LongChair** [comment]
is this still an issue with the latest builds ?

**Comment by Lanthade** [comment]
Yes, just tested sending a select command against 2.8.0.813-6e77dc0c on OSX 10.11

My method is to send a poll command and then to send the control command.

Poll command:

`"GET /player/timeline/poll?commandID=0&X-Plex-Device-Name="..dev_name.." HTTP/1.1\r\nX-Plex-Client-Identifier: "..client_id.."\r\n\r\n"`

And select command:

`"GET /player/navigation/select?commandID="..cmdID.."&X-Plex-Device-Name="..dev_name.." HTTP/1.1\r\nX-Plex-Client-Identifier: "..client_id.."\r\n\r\n"`

Command is sent immediately after opening the program, crashes every single time.

When I first posted the problem I was working with the subscription connection model but I switched to polling because that is a much better fit for my application.

**Comment by LongChair** [comment]
ok thanks for the feedback, can we close this then ?

**Comment by Lanthade** [comment]
@LongChair The code snippet above is in Lua and is coming from a QSC Q-SYS Core.  I'll convert it to perl and see if the issue persists.  If so I'll post up that code since that'll be a lot easier.  If it only happens from the Q-SYS stuff I'll throw together a simple design and hook you up with the software and instructions on how to run the design in emulator mode.

**Comment by LongChair** [comment]
@Lanthade : i appreciate your efforts.
If you can provide me with a simple way to reproduce, (maybe curl would work) then i can have a look :)

**Comment by LongChair** [comment]
@Lanthade : any news on this ? :)
